pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -359,14 +359,14 @@ stdenv.mkDerivation {
     ''
     + optionalString (libcxx == null && (useGccForLibs && gccForLibs.langCC or false)) ''
       for dir in ${gccForLibs}${lib.optionalString (hostPlatform != targetPlatform) "/${targetPlatform.config}"}/include/c++/*; do
-        echo "-isystem $dir" >> $out/nix-support/libcxx-cxxflags
+        echo "-cxx-isystem $dir" >> $out/nix-support/libc-cflags
       done
       for dir in ${gccForLibs}${lib.optionalString (hostPlatform != targetPlatform) "/${targetPlatform.config}"}/include/c++/*/${targetPlatform.config}; do
-        echo "-isystem $dir" >> $out/nix-support/libcxx-cxxflags
```

Comment:
Note that these flags are not just consumed by clang but also by some other tools i.e. bindgen for rust.
I think most of these third-party tools are llvm-based but we should at least check if they also support this prefix:

- bindgen
- ccls
- clang-tools/clangd (very likely)

Reply:
You're right. I'll add a commit later to add some patches for tools using libcxx-cxxflags.